commcare_export/minilinq.py

```diff
@@ -428,31 +431,37 @@ def coerce_row(self, row):
 
     def eval(self, env):
         rows = self.source.eval(env)
-        env.emit_table({'name': self.table,
-                        'headings': [heading.eval(env) for heading in self.headings],
-                        'rows': map(self.coerce_row, rows)})
+        env.emit_table(TableSpec(
+            name=self.table,
+            headings=[heading.eval(env) for heading in self.headings],
+            rows=map(self.coerce_row, rows),
+        ))
 
     @classmethod
     def from_jvalue(cls, jvalue):
         fields = jvalue['Emit']
-
-        return cls(table    = fields['table'],
-                   source   = MiniLinq.from_jvalue(fields['source']),
-                   headings = [MiniLinq.from_jvalue(heading) for heading in fields['headings']],
-                   missing_value=fields.get('missing_value'))
+        return cls(
+            table=fields['table'],
+            source=MiniLinq.from_jvalue(fields['source']),
+            headings=[MiniLinq.from_jvalue(heading) for heading in fields['headings']],
+            missing_value=fields.get('missing_value'),
+            data_types=fields.get('data_types'),
+        )
 
     def to_jvalue(self):
         return {'Emit': {'table': self.table,
                          'headings': [heading.to_jvalue() for heading in self.headings],
                          'source': self.source.to_jvalue(),
-                         'missing_value': self.missing_value}}
+                         'missing_value': self.missing_value,
+                         'data_types': self.data_types}}
```

Comment:
I noticed that this breaks --dump-query. The data_types should be to_jvalue'd like headings are.

Reply:
nice catch! https://github.com/dimagi/commcare-export/pull/142/commits/47c96325813297d5d909a323ec3027e07d7bf85a